components/script/dom/gpusupportedfeatures.rs

```diff
@@ -39,43 +39,66 @@ pub struct GPUSupportedFeatures {
     // internal storage for features
     #[custom_trace]
     internal: DomRefCell<IndexSet<GPUFeatureName>>,
+    #[ignore_malloc_size_of = "defined in wgpu-types"]
+    #[no_trace]
+    features: Features,
 }
 
 impl GPUSupportedFeatures {
     fn new(
         global: &GlobalScope,
         proto: Option<HandleObject>,
-        features: wgt::Features,
+        features: Features,
     ) -> DomRoot<GPUSupportedFeatures> {
         let mut set = IndexSet::new();
-        if features.contains(wgt::Features::DEPTH_CLIP_CONTROL) {
+        if features.contains(Features::DEPTH_CLIP_CONTROL) {
             set.insert(GPUFeatureName::Depth_clip_control);
         }
-        if features.contains(wgt::Features::DEPTH32FLOAT_STENCIL8) {
+        if features.contains(Features::DEPTH32FLOAT_STENCIL8) {
             set.insert(GPUFeatureName::Depth32float_stencil8);
         }
-        if features.contains(wgt::Features::PIPELINE_STATISTICS_QUERY) {
-            set.insert(GPUFeatureName::Pipeline_statistics_query);
-        }
-        if features.contains(wgt::Features::TEXTURE_COMPRESSION_BC) {
+        if features.contains(Features::TEXTURE_COMPRESSION_BC) {
             set.insert(GPUFeatureName::Texture_compression_bc);
         }
-        if features.contains(wgt::Features::TEXTURE_COMPRESSION_ETC2) {
+        // TODO: texture-compression-bc-sliced-3d when wgpu supports it
+        if features.contains(Features::TEXTURE_COMPRESSION_ETC2) {
             set.insert(GPUFeatureName::Texture_compression_etc2);
         }
-        if features.contains(wgt::Features::TEXTURE_COMPRESSION_ASTC) {
+        if features.contains(Features::TEXTURE_COMPRESSION_ASTC) {
             set.insert(GPUFeatureName::Texture_compression_astc);
         }
-        if features.contains(wgt::Features::TIMESTAMP_QUERY) {
+        if features.contains(Features::TIMESTAMP_QUERY) {
             set.insert(GPUFeatureName::Timestamp_query);
         }
-        if features.contains(wgt::Features::INDIRECT_FIRST_INSTANCE) {
+        if features.contains(Features::INDIRECT_FIRST_INSTANCE) {
             set.insert(GPUFeatureName::Indirect_first_instance);
         }
+        // While this feature exists in wgpu, it's not supported by naga yet
+        // https://github.com/gfx-rs/wgpu/issues/4384
+        /*
+        if features.contains(Features::SHADER_F16) {
+            set.insert(GPUFeatureName::Shader_f16);
+        }
+        */
+        if features.contains(Features::RG11B10UFLOAT_RENDERABLE) {
+            set.insert(GPUFeatureName::Rg11b10ufloat_renderable);
+        }
+        if features.contains(Features::BGRA8UNORM_STORAGE) {
+            set.insert(GPUFeatureName::Bgra8unorm_storage);
+        }
+        if features.contains(Features::FLOAT32_FILTERABLE) {
+            set.insert(GPUFeatureName::Float32_filterable);
+        }
+        // TODO: clip-distances when wgpu supports it
```

Comment:
For consistency's sake, this todo could have the same format as the others added in this PR.

Reply:
idk what you mean (the sagudev part?)